### ISSUE TITLE
how to train my own dataset and use pretrained params?

### ISSUE THREAD

**Issue by fouadchacccour** [dmlc/gluon-cv#707]
Hello, I'm new in computer vision field. I'm using gluoncv to do instance segmentation with mask-rcnn but i have some problems.
I did my own dataset similar to coco format and I trained with this model "mask_rcnn_resnet50_v1b_coco", but my dataset has just 2 classes. After training i got the parameter file and when i loaded the parameters and predict a new image, the classification was wrong and still loaded all classes of the coco dataset (80 classes). So is there any solution to customize the number of classes?

**Comment by fouadchacccour** [comment]
Hello, I'm new in computer vision field. I'm using gluoncv to do instance segmentation with mask-rcnn but i have some problems.
I did my own dataset similar to coco format and I trained with this model "mask_rcnn_resnet50_v1b_coco", but my dataset has just 2 classes. After training i got the parameter file and when i loaded the parameters and predict a new image, the classification was wrong and still loaded all classes of the coco dataset (80 classes). So is there any solution to customize the number of classes?

**Comment by zx-code123** [comment]
Are you sure train with your own data ,it seems like you still train with coco dataset?

**Comment by fouadchacccour** [comment]
I have changed the model "mask_rcnn_resnet50_v1b_coco" to handle my dataset and it works but if i do the training from scratch. If I use pretrained weights I get an error because there is a mismatch with the trained classes and my own. Is there any way to use the pretrained network to upgrade the weights with my own classes @zx-code123

**Comment by zx-code123** [comment]
Actually,I do not understand what you mean.However,if you want to train with your own data,you should change some code in [instance.py](https://github.com/dmlc/gluon-cv/blob/master/gluoncv/data/mscoco/instance.py) ,for example change CLASSES for coco to yours and the dataset path.

**Comment by fouadchacccour** [comment]
Thank you for your reply,
I did this and I changed the classes and the path of my dataset in instance.py but the problem is that i can not do the fine-tuning and use the pretrained weights or the pretrained model of coco. I used the function "net.reset_class(['class1','class2'])" in the inference code and it doesn't work i got this Error: 
"Cannot reset context for Parameter 'maskrcnn0_maskrcnn0_dense0_weight' because it has not been initialized."
Do you have an idea how can I use the pretrained weights and do the fine-tuning on my own dataset ?@zx-code123

**Comment by zx-code123** [comment]
I try use "net.reset_class(['class1','class2'])" with pretrained model of coco,it is worked.However,when i inference the image ,the result is bad.And i find https://github.com/dmlc/gluon-cv/issues/716#issuecomment-479120039.         So,you can only use it  to finetune your dataset.

**Comment by mohamadassaf96** [comment]
close #707
